test/integration/gardenlet/seed/seed/seed_test.go

```diff
@@ -333,25 +364,27 @@ var _ = Describe("Seed controller tests", func() {
 					By("Delete Seed")
 					Expect(testClient.Delete(ctx, seed)).To(Succeed())
 
-					By("Verify that the seed system components have been deleted")
-					Eventually(func(g Gomega) []resourcesv1alpha1.ManagedResource {
-						managedResourceList := &resourcesv1alpha1.ManagedResourceList{}
-						g.Expect(testClient.List(ctx, managedResourceList, client.InNamespace(testNamespace.Name))).To(Succeed())
-						return managedResourceList.Items
-					}).Should(BeEmpty())
+					if seedIsGarden {
```

Comment:
This is already ensured by the controller,
https://github.com/gardener/gardener/blob/9634d37e9ba8fd8ef54be9fa7dd997856f335b54/pkg/gardenlet/controller/seed/seed/reconciler_delete.go#L297-L330

Reply:
I'm confused why you added this `if` block.
Shouldn't the controller cleanup all created `ManagedResources` in both cases?

Reply:
Because if seedIsGarden, `resourcemanager.Destroy` is not called, and this function cleans up the CRDs.
When I try to assert the same when seedIsNotGarden, then before we check for the ManagedResourceList to be empty, the controller has already cleanedup the CRDs and it gives,
```
              Message: "the server could not find the requested resource (get managedresources.resources.gardener.cloud)",
```
like in the prow logs above.

Reply:
Got it, thanks for explaining. Can you add a corresponding code comment to explain this to others reading the code?

Reply:
Done with https://github.com/gardener/gardener/pull/7044/commits/c1baa147dff6213c967c605277fc12f80fc6bfe9, PTAL!